app/models/profile.rb

```diff
@@ -70,10 +70,11 @@ def clone_to(account: nil, host: nil, external: true)
       (new_profile = dup).update!(account: account, hosts: [host],
                                   parent_profile: self,
                                   external: external)
+      new_profile.update_rules(ref_ids: rules.pluck(:ref_id))
     else
       new_profile.hosts << host unless new_profile.hosts.include?(host)
     end
-    new_profile.update_rules(ref_ids: rules.pluck(:ref_id))
+
```

Comment:
This is it. Never update rules on existing profiles. The TestResult will still show all relevant rules, even outside of the associated profile, but the profile rules themselves should never be changed.